src/controllers/qasm_controller.hpp

```diff
@@ -753,8 +753,9 @@ void QasmController::set_parallelization_circuit(
     case Method::statevector_thrust_cpu:
     case Method::stabilizer:
     case Method::matrix_product_state: {
-      if ((noise_model.is_ideal() || !noise_model.has_quantum_errors()) &&
-          check_measure_sampling_opt(circ, Method::statevector).first) {
+      if (circ.shots == 1 || (
+          noise_model.has_quantum_errors() == false &&
```

Comment:
I think is more readable `!noise_model.has_quantum_errors()` instead of `noise_model.has_quantum_errors() == false`. What do you think?